src/query/xasl.h

```diff
@@ -605,7 +605,7 @@ struct cte_proc_node
 	      if ((_x)->status == XASL_CLEARED || (_x)->status == XASL_INITIALIZED) \
 		{ \
 		  /* execute xasl query */ \
-		  if (_x->sub_xasl_id) \
+		  if (_x->sub_xasl_id  && _x->single_tuple) \
```

Comment:
아래처럼 assert를 하나 걸어 두는 것이 어떤가요?

```suggestion
assert( (_x->single_tuple && _x->is_single_tuple) || (_x->single_tuple  == NULL && _x->is_single_tuple == 0)) ;  \
if (_x->sub_xasl_id && _x->single_tuple) \
```